### PR TITLE
[Lake] Add LakeTabletsChannel

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

### CODE REVIEW

File: be/src/runtime/lake_tablets_channel.cpp [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    context->_channel_row_idx_start_points = std::make_unique<uint32_t[]>(channel_size + 1);
+
+    auto& row_indexes = context->_row_indexes;
+    auto& channel_row_idx_start_points = context->_channel_row_idx_start_points;
+
+    // compute row indexes for each channel
+    for (uint32_t i = 0; i < request.tablet_ids_size(); ++i) {
+        uint32_t channel_index = _tablet_id_to_sorted_indexes[request.tablet_ids(i)];
+        channel_row_idx_start_points[channel_index]++;
+    }
```
Comment (imay) [inline be/src/runtime/lake_tablets_channel.cpp:422]: this could also use the following tablet_ids and tablet_ids_size

**Comment by sduzh** [comment]
rin starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_clang-format

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_fe_unittest

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Review by wuleistarrocks — DISMISSED** [review]

File: be/src/runtime/lake_tablets_channel.cpp [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+                return;
+            }
+            std::lock_guard l(_mtx);
+            if (_response->status().status_code() == TStatusCode::OK) {
+                status.to_protobuf(_response->mutable_status());
+            }
+        }
+
+        void add_finished_tablet(int64_t tablet_id) {
+            DCHECK(_response != nullptr);
```
Comment (wyb) [inline be/src/runtime/lake_tablets_channel.cpp:87]: When is `_response` nullptr?

Comment (sduzh) [reply be/src/runtime/lake_tablets_channel.cpp:87]: Never

Comment (wyb) [reply be/src/runtime/lake_tablets_channel.cpp:87]: Better to remove this dcheck.

Comment (sduzh) [reply be/src/runtime/lake_tablets_channel.cpp]: updated

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**